### ISSUE TITLE
Move Pin Button Into Context Menu

### ISSUE THREAD

**Issue by grgia** [Expensify/App#22635]
Part of https://github.com/Expensify/App/issues/20486#issuecomment-1622110749

Relevant:
https://github.com/Expensify/App/issues/22332#issuecomment-1628887301
>Yeah, I left a similar comment over in the other issue but I think we should stay consistent with the 
> overflow menu on all of the various report type views (chat, thread, task, report, expense, etc). And then in the very least, it always has a pin option but it also might have:
> 
> call
> delete
> etc

**Comment by grgia** [comment]
Relevant Convo: https://expensify.slack.com/archives/C02HWMSMZEC/p1687294634568589?thread_ts=1686946331.451439&cid=C02HWMSMZEC

**Comment by grgia** [comment]
I looked into this, and it's going to require some refactoring. I think we should handle this similarly to the report action context menus where we display different options depending on the reportAction (except here, we would display different options depending on the report)

**Comment by grgia** [comment]
@JmillsExpensify could you please help me locate the desired pin button / three dots menu item behavior for all report types?

**Comment by JmillsExpensify** [comment]
Apologies, just seeing this one.

**Comment by JmillsExpensify** [comment]
This is a bit tricky, since some of these haven't been mocked up. I'll go ahead and assign myself and come back to this.

**Comment by grgia** [comment]
Should we hold this? @JmillsExpensify

**Comment by JmillsExpensify** [comment]
@shawnborton @dannymcclain Can one of you take this on by chance? I haven't been able to get to it.

**Comment by shawnborton** [comment]
Sure, let's spin the wheel!

**Comment by shawnborton** [comment]
Maybe we could combine the call options into the same menu with pin?
<img width="966" alt="image" src="https://github.com/Expensify/App/assets/2319350/6c815026-ed77-459f-9c24-42055f376b49">

![image](https://github.com/Expensify/App/assets/2319350/5aaf7dc7-7010-42bc-a5e1-4256e0b1d606)



cc @trjExpensify as well

**Comment by trjExpensify** [comment]
Yeah, I think that works instead of an additional level. Once we have in-app calling & screenshare built, I think we remove the Zoom and Google Meet options completely.

**Comment by JmillsExpensify** [comment]
Nice, I agree on this path, including deprecating Meet and Zoom once we support in-app calling.

**Comment by grgia** [comment]
@JmillsExpensify can I get a priority gut check for this?

**Comment by trjExpensify** [comment]
With the design hashed out, can we make it `external`?

**Comment by grgia** [comment]
@trjExpensify good shout, I don't see why not.

![image](https://github.com/Expensify/App/assets/38015950/a2e91167-e27e-4c2e-a473-5ceafc6a99d9)

![image](https://github.com/Expensify/App/assets/38015950/0870eadf-950a-4391-8cb4-9f1c615fd3a4)

Can we confirm that the above screenshots are the final design with options:
- Pin
- Zoom
- Google Meet

And then we need to know:
- Which types of reports show these options
- What happens on click for all of the options

**Comment by shawnborton** [comment]
Actually I think we have some overlap here, pretty sure @srikarparsi submitted a PR for this already in relation to some thread/subscribe/unsubscribe changes?

**Comment by srikarparsi** [comment]
Oh yup right here: https://github.com/Expensify/App/pull/27748

**Comment by trjExpensify** [comment]
Dope, let's close this then or link that PR to this issue perhaps?
